### DISCUSSION TITLE
Redirects from Flux domain to flux releases

### DISCUSSION THREAD

**Discussion by alisondy** [fluxcd/flux2#1798] [Proposals]
Currently to directly curl a Flux release, you would a url like this

```
https://github.com/fluxcd/flux2/releases/download/v0.17.0/flux_0.17.0_linux_arm.tar.gz
```

## Proposal

I want to curl a url like the following to grab a Flux release

```
https://fluxcd.io/releases/v0.0.17/linux/amd
```

.. or the latest release

```
https://fluxcd.io/releases/latest/linux/amd
```

This url would redirect to the corresponding file on GitHub.

## Benefits

- Easier to remember where to download flux
- Can deprecate the install script with documentation
		- curl url
			- Replace parameters according to your OS
		- unzip it
		- optional checksum
		- run install/mv on the file extracted
- Don't have to update download link on docs when a new release is published

## Work involved

- Create a GitHub action that updates Netlify redirect config in website repo. This action would trigger when a new release is published
	- Add line to file with redirect
	- Update line with latest to point to new release

## Other notes

This is a minor quality of life improvement, not urgent.

Being able to download the binary directly rather than in a tarred form would make installation process shorter. 

Changing the release proccess to publish a binary should be covered in another proposal, after this one is implemented.

**Comment by stefanprodan** [comment]
The install script does a lot of things, most notable it verifies the checksum of the binary, using just curl for installing should never be case. We should also encourage people to use a package manager instead of using using the install script.
